plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java

```diff
@@ -341,6 +394,11 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(C
     {
         MongoTableHandle handle = (MongoTableHandle) table;
 
+        if (isPassthroughQuery(handle)) {
```

Comment:
Can we try to perform an AND of the filter we inject via PTF and the filter we apply in SQL.